bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigUtil.java

```diff
@@ -63,9 +67,49 @@ public static Object normalizeType(Object value, ConfigDescriptionParameter conf
         if (configDescriptionParameter != null) {
             Normalizer normalizer = NormalizerFactory.getNormalizer(configDescriptionParameter);
             return normalizer.normalize(value);
-        } else {
-            return value instanceof Double ? BigDecimal.valueOf((Double) value) : value;
+        } else if (value == null || value instanceof Boolean || value instanceof String
+                || value instanceof BigDecimal) {
+            return value;
+        } else if (value instanceof Number) {
+            return new BigDecimal(value.toString());
```

Comment:
Is the toString() always precise and ok regarding performance? Before, Doubles were directly wrapped into the BigDecimal as a Double (without doing a string serialization inbetween). Might it be useful to do more specific BigDecimal constructions depending on the value type?